pkg/apiserver/store/vpc_inventory.go

```diff
@@ -0,0 +1,160 @@
+	if nodeName, found := selectors.Field.RequiresExactMatch("nodeName"); found {
+		if !obj.(types.Span).Has(nodeName) {
+			return false
+		}
+	}
+	return true
+}
+
+// isSelected determines if the previous and the current version of an object should be selected by the given selectors.
+func isSelected(key string, prevObj, currObj interface{}, selectors *storage.Selectors, isInitEvent bool) (bool, bool) {
+	// We have filtered out init events that we are not interested in, so the current object must be selected.
+	if isInitEvent {
+		return false, true
+	}
+	prevObjSelected := !reflect.ValueOf(prevObj).IsNil() && keyAndSpanSelectFunc(selectors, key, prevObj)
+	currObjSelected := !reflect.ValueOf(currObj).IsNil() && keyAndSpanSelectFunc(selectors, key, currObj)
+	return prevObjSelected, currObjSelected
+}
+
+// ToWatchEvent converts the vpcEvent to *watch.Event based on the provided Selectors. It has the following features:
+// 1. Added event will be generated if the Selectors was not interested in the object but is now.
+// 2. Modified event will be generated if the Selectors was and is interested in the object but is now.
+// 3. Deleted event will be generated if the Selectors was interested in the object but is not now.
+func (event *vpcInventoryEvent) ToWatchEvent(selectors *storage.Selectors, isInitEvent bool) *watch.Event {
+	prevObjSelected, currObjSelected := isSelected(event.Key, event.PrevObject, event.CurrObject, selectors, isInitEvent)
+	switch {
+	case !currObjSelected && !prevObjSelected:
+		return nil
+	case currObjSelected && !prevObjSelected:
+		// Watcher was not interested in that object but is now, an added event will be generated.
+		return &watch.Event{Type: watch.Added, Object: event.CurrObject}
+	case currObjSelected && prevObjSelected:
+		// Watcher was not interested in that object but is now, an added event will be generated.
+		return &watch.Event{Type: watch.Modified, Object: event.CurrObject}
+	case !currObjSelected && prevObjSelected:
+		// Watcher was interested in that object but is not interested now, a deleted event will be generated.
+		return &watch.Event{Type: watch.Deleted, Object: event.PrevObject}
+	}
+	return nil
+}
+
+func (event *vpcInventoryEvent) GetResourceVersion() uint64 {
+	return event.ResourceVersion
+}
+
+var _ storage.GenEventFunc = genVPCEvent
+
+// genVPCEvent generates InternalEvent from the given versions of an VPC.
+func genVPCEvent(key string, prevObj, currObj interface{}, rv uint64) (storage.InternalEvent, error) {
+	if reflect.DeepEqual(prevObj, currObj) {
+		return nil, nil
+	}
+
+	event := &vpcInventoryEvent{Key: key, ResourceVersion: rv}
+	if prevObj != nil {
+		event.PrevObject = prevObj.(*runtimev1alpha1.Vpc)
+	}
+
+	if currObj != nil {
+		event.CurrObject = currObj.(*runtimev1alpha1.Vpc)
+	}
+
+	return event, nil
+}
+
+// VPCKeyFunc knows how to get the key of an VPC.
+func VPCKeyFunc(obj interface{}) (string, error) {
```

Comment:
Change it to vpcKeyFunc, as it is not referenced outside the package.

Reply:
changed